engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessService.scala

```diff
@@ -60,27 +60,27 @@ class PeriodicProcessService(
 
   def getScenarioActivitiesSpecificToPeriodicProcess(
       processIdWithName: ProcessIdWithName
-  ): Future[List[ScenarioActivity]] =
-    scheduledProcessesRepository
-      .getSchedulesState(processIdWithName.name)
-      .run
-      .map(_.groupedByPeriodicProcess)
-      .map(_.flatMap(_.deployments))
-      .map(_.map { deployment =>
-        ScenarioActivity.PerformedScheduledExecution(
-          scenarioId = ScenarioId(processIdWithName.id.value),
-          scenarioActivityId = ScenarioActivityId.random,
-          user = ScenarioUser.internalNuUser,
-          date = instantAtUTC(deployment.runAt),
-          scenarioVersionId = Some(ScenarioVersionId.from(deployment.periodicProcess.processVersion.versionId)),
-          dateFinished = deployment.state.completedAt.map(instantAtUTC),
-          scheduleName = deployment.scheduleName.display,
-          status = scheduledExecutionStatus(deployment.state.status),
-          createdAt = instantAtUTC(deployment.createdAt),
-          nextRetryAt = deployment.nextRetryAt.map(instantAtUTC),
-          retriesLeft = deployment.nextRetryAt.map(_ => deployment.retriesLeft),
-        )
-      }.toList.sortBy(_.date))
+  ): Future[List[ScenarioActivity]] = for {
+    schedulesState <- scheduledProcessesRepository.getSchedulesState(processIdWithName.name).run
+    groupedByProcess        = schedulesState.groupedByPeriodicProcess
+    deployments             = groupedByProcess.flatMap(_.deployments)
+    deploymentsWithStatuses = deployments.flatMap(d => scheduledExecutionStatus(d.state.status).map((d, _)))
+    activities = deploymentsWithStatuses.map { case (deployment, status) =>
+      ScenarioActivity.PerformedScheduledExecution(
+        scenarioId = ScenarioId(processIdWithName.id.value),
+        scenarioActivityId = ScenarioActivityId.random,
+        user = ScenarioUser.internalNuUser,
+        date = instantAtUTC(deployment.runAt),
+        scenarioVersionId = Some(ScenarioVersionId.from(deployment.periodicProcess.processVersion.versionId)),
+        dateFinished = deployment.state.completedAt.map(instantAtUTC),
+        scheduleName = deployment.scheduleName.display,
+        scheduledExecutionStatus = status,
+        createdAt = instantAtUTC(deployment.createdAt),
+        nextRetryAt = deployment.nextRetryAt.map(instantAtUTC),
+        retriesLeft = deployment.nextRetryAt.map(_ => deployment.retriesLeft),
+      )
+    }
+  } yield activities
```

Comment:
do we miss the sort?

Reply:
Activities from both sources (DB and DeploymentManager) are merged in the HttpService class, they are sorted there. In this method there is no guarantee (both in method name and return type, it is not a sorted list/set) that the result is sorted. That way it is done only once.